### ISSUE TITLE
Unstable training loops, exploding loss values

### ISSUE THREAD

**Issue by MicheleCattaneo** [ecmwf/anemoi-core#847]
### What happened?

There are multiple reports of training instability, where the training loss displays sudden jumps that were not observed previously.
Below an example using anemoi with [this release](https://github.com/ecmwf/anemoi-core/commit/120986f645a30318cb5b330c58a99edbc1dd07a7), i.e training: v0.8.4, models: v0.11.3, graphs: v0.8.3.

<img width="492" height="361" alt="Image" src="https://github.com/user-attachments/assets/92e38d9e-bff5-46bf-8819-e4856c940782" />

However, the issue has appeared in previous releases as well.

The issue appears to be (mostly?) happening for global models

### What are the steps to reproduce the bug?

A simple training run with default configurations can reproduce the issue.

### Version

training: v0.8.4, models: v0.11.3, graphs: v0.8.3

### Platform (OS and architecture)

Linux, A100 Nvidia GPUs

### Relevant log output

```shell

```
